gravitee-io/gravitee-access-management
# fix: encoding the clientId in overview so the login flow can be initiated correctly

## :id: Reference related issue. 

https://github.com/gravitee-io/issues/issues/8808

## :pencil2: A description of the changes proposed in the pull request

This PR encodes the clientId in the Login Flow URL presented by the Overview page

It also ensure that in the email template, the client_id is encoded correctly

## :memo: Test scenarios 

- Create am application with characters that needs URI encoding (`my # client-app`)
Test 1:
- Go to `Applications > Your App > Overview` and try the  `Initiate the login flow` and login with you actual user, you should be able to login with the client id being encoded

Test 2: (separately)
- Try the forgot password in order for the user to receive an email with reset password
- Try creating a user with registration confirmation email
These two last cases should have a link where the client_id is encoded
